File: PACMAN_Java/src/resources/Entity.java
package resources;

public class Entity {
    private short x,y,lives;//short limits from -32768 to 32767
    private char symbol;// char limits from 0 to 65535


    //constructors
    public Entity(){
        this.x = 0;
        this.y = 0;
        this.lives = 1;
        this.symbol = '%';
    }

    public Entity(short x, short y, short lives, char symbol){
        this.x = x;
        this.y = y;
        this.lives = lives;
        this.symbol = symbol;
    }


    //setters
    public void setX(short x) {
        this.x = x;
    }

    public void setY(short y) {
        this.y = y;
    }

    public void setLives(short lives) {
        this.lives = lives;
    }

    public void setSymbol(char symbol) {
        this.symbol = symbol;
    }


    //Getters
    public short getX() {
        return x;
    }

    public short getY() {
        return y;
    }

    public short getLives() {
        return lives;
    }

    public char getSymbol() {
        return symbol;
    }

    public void setXY(short x, short y){

    }

    public void display(){

    }

    public void display(char symbol){

    }

    public void replace(){

    }

    public void move_up(){}
    public void move_down(){}
    public void move_left(){}
    public void move_right(){}
}

File: PACMAN_Cpp/src/Entity.cpp
#include "Entity.h"
Entity::Entity()
{
	x = 0;
	y = 0;
	lives = 1;
	symbol = '%';
}

Entity::Entity(short X, short Y, short L, char S)
{
	x = X;
	y = Y;
	lives = L;
	symbol = S;
}

void Entity::setX(short val)
{
	x = val;
}

void Entity::setY(short val)
{
	y = val;
}

void Entity::setLives(short val)
{
	lives = val;
}

void Entity::setSymbol(char val)
{
	symbol = val;
}

short Entity::getX() const
{
	return x;
}

short Entity::getY() const
{
	return y;
}

short Entity::getLives() const
{
	return lives;
}

char Entity::getSymbol() const
{
	return symbol;
}

void Entity::gotoxy(short i, short j) const
{
	HANDLE hStdout = GetStdHandle(STD_OUTPUT_HANDLE);
	COORD position = { i, j };
	SetConsoleCursorPosition(hStdout, position);
}
void Entity::display() const {
	gotoxy(x, y);
	std::cout << symbol;
}

void Entity::display(char c) const {
	gotoxy(x, y);
	std::cout << c;
}

void Entity::replace() const {
	gotoxy(x, y);
}
